libs/barista-components/chart/src/tooltip/chart-tooltip.ts

```diff
@@ -165,7 +153,13 @@ export class DtChartTooltip implements OnDestroy {
       const positionStrategy = this._overlay
         .position()
         .flexibleConnectedTo(
-          this._getTooltipPosition(data, parentChart, plotBackgroundInfo),
+          this.customPositioner?.positionFunction !== undefined
+            ? this.customPositioner.positionFunction(
+                data,
+                parentChart,
+                plotBackgroundInfo,
+              )
+            : getDefaultTooltipPosition(data, parentChart, plotBackgroundInfo),
```

Comment:
hmm, you are repeating logic here. Maybe it would be a good idea to extract that logic into a member function, to only maintain it once.

Reply:
This method is a bit different from the previous one and it is called when updating the chart.
I can extract a method getStrategy() in order to avoid the duplication of this check